internal/cli/base.go

```diff
@@ -306,6 +306,9 @@ func (c *baseCommand) DoApp(ctx context.Context, f func(context.Context, *client
 		if err := f(ctx, app); err != nil {
 			if err != ErrSentinel {
 				finalErr = multierror.Append(finalErr, err)
+			} else {
+				// if we have an ErrSentinel here, we've already output info to the UI
+				return err
```

Comment:
I think what we should do here is continue to append to finalErr, but note we got a sentinel (switch some bool to true), then at the end something like:

```go
if finalErr == nil && didErrSentinel {
  finalErr = ErrSentinal
}

return finalErr
```

The reason being: this way we attempt to run against other apps here before we exit. The idea of this function one day is that we can try multiple apps in parallel maybe (its not today).